core/src/main/java/org/apache/iceberg/util/SortOrderUtil.java

```diff
@@ -64,4 +68,16 @@ static SortOrder buildSortOrder(Schema schema, PartitionSpec spec, SortOrder sor
 
     return builder.build();
   }
+
+  public static Set<String> getSortedColumns(SortOrder sortOrder) {
+    if (sortOrder == null) {
```

Comment:
Do we need this extra check for null here? Looks like we validate the order is not null before calling. I'd probably simplify  and assume it is not null. It is a pretty internal utility class.

Reply:
Or keep the check here but remove in the caller. Up to you.

Reply:
Done, removed in caller and kept null check in the method.